src/Symfony/Component/AssetMapper/MappedAsset.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper;
+
+/**
+ * Represents a single asset in the asset mapper system.
+ *
+ * @experimental
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class MappedAsset
+{
+
+    private string $publicPath;
+    private string $sourcePath;
+    private string $content;
+    private string $digest;
+    private bool $isPredigested;
+    private ?string $mimeType;
+    /** @var AssetDependency[]  */
+    private array $dependencies = [];
+
+    public function __construct(private string $logicalPath)
```

Comment:
I'm wondering if we couldn't make all the properties of this class `public readonly`, and remove/most setters+getters?

Reply:
Without accessors, it is a lot harder to write a BC layer for properties than for methods though.